protocol/messenger.go

```diff
@@ -241,6 +243,7 @@ func NewMessenger(
 	installationID string,
 	peerStore *mailservers.PeerStore,
 	accountsManager *account.GethManager,
+	rpcClient *rpc.Client,
 	opts ...Option,
```

Comment:
Here is the use of `Option` structs in `NewMessenger`, this prevents the bloating of the function signature and allows for a more configurable struct with `Option`al parameters.